lib/ansible/template/safe_eval.py

```diff
@@ -123,13 +129,14 @@ def generic_visit(self, node, inside_call=False):
         else:
             return result
     except SyntaxError as e:
+        display.warning('SyntaxError in safe_eval() on expr: %s (%s)' % (expr, e))
```

Comment:
I am not sure that this merits displaying error - as per the note below, there is some special handling for this exception that results in passing the string back as-is for later processing. I appear to have been relying on this behavior, as I am now getting a ton of SyntaxError warnings on Jinja2 templates inlined in my group_vars YAML.